File: frontend_v2/src/services/labelsService.js
import api from './api';

class LabelsService {
  constructor() {
    this.labels = {};
    this.loaded = false;
  }

  // Load all labels from backend
  async loadLabels() {
    if (this.loaded && Object.keys(this.labels).length > 0) {
      return this.labels;
    }

    try {
      const labels = await api.get('/api/labels');
      this.labels = labels || {};
      this.loaded = true;
      return this.labels;
    } catch (error) {
      console.error('Failed to load labels:', error);
      // Return empty object if loading fails
      return {};
    }
  }

  // Get a specific label by key
  getLabel(key, defaultValue = null) {
    return this.labels[key] || defaultValue || key;
  }

  // Get all labels
  getAllLabels() {
    return this.labels;
  }

  // Check if labels are loaded
  isLoaded() {
    return this.loaded;
  }

  // Clear labels (useful for reloading)
  clear() {
    this.labels = {};
    this.loaded = false;
  }
}

export default new LabelsService();

File: event-planning-service/src/main/java/com/vibeworks/planning/service/LabelsService.java
package com.vibeworks.planning.service;

import lombok.extern.slf4j.Slf4j;
import org.springframework.core.io.ClassPathResource;
import org.springframework.stereotype.Service;

import java.io.IOException;
import java.io.InputStream;
import java.util.HashMap;
import java.util.Map;
import java.util.Properties;

@Service
@Slf4j
public class LabelsService {
    
    private final Map<String, String> labels = new HashMap<>();
    
    public LabelsService() {
        loadLabels();
    }
    
    private void loadLabels() {
        try {
            ClassPathResource resource = new ClassPathResource("labels.properties");
            Properties properties = new Properties();
            
            try (InputStream inputStream = resource.getInputStream()) {
                properties.load(inputStream);
            }
            
            // Convert Properties to Map
            for (String key : properties.stringPropertyNames()) {
                labels.put(key, properties.getProperty(key));
            }
            
            log.info("Loaded {} labels from properties file", labels.size());
        } catch (IOException e) {
            log.error("Failed to load labels.properties file", e);
        }
    }
    
    public Map<String, String> getAllLabels() {
        return new HashMap<>(labels);
    }
    
    public String getLabel(String key) {
        return labels.getOrDefault(key, key);
    }
    
    public String getLabel(String key, String defaultValue) {
        return labels.getOrDefault(key, defaultValue);
    }
    
    public Map<String, String> getLabelsByPrefix(String prefix) {
        Map<String, String> filtered = new HashMap<>();
        for (Map.Entry<String, String> entry : labels.entrySet()) {
            if (entry.getKey().startsWith(prefix)) {
                filtered.put(entry.getKey(), entry.getValue());
            }
        }
        return filtered;
    }
}

